### PR TITLE
Persistent bitmaps removal test

### PR DESCRIPTION
**Pull request by zhencliu**
ID: 1905322

### CODE REVIEW

**Comment by zhencliu** [comment]
local fs:
 (1/1) type_specific.io-github-autotest-qemu.blockdev_inc_backup_rm_persistent_bitmap: PASS (139.77 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-12-11T06.42-a69f344/results.html
JOB TIME   : 142.59 s

iscsi-direct + virtio-blk:
 (1/1) type_specific.io-github-autotest-qemu.blockdev_inc_backup_rm_persistent_bitmap: PASS (113.06 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-12-12T05.24-8beb5ec/results.html
JOB TIME   : 116.02 s

ceph:
 (1/1) type_specific.io-github-autotest-qemu.blockdev_inc_backup_rm_persistent_bitmap: PASS (168.96 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-12-12T05.36-5be3e30/results.html
JOB TIME   : 171.67 s

**Comment by zhencliu** [comment]
For nbd backend, I hit the following issue:
01:55:45 DEBUG| Send command: {'execute': 'system_powerdown', 'id': 'WBBKmX4m'}
01:55:45 INFO | POWERDOWN QMP event received
01:55:54 INFO | [qemu output] qemu-kvm: Lost persistent bitmaps during inactivation of node 'drive_data1': Failed to write bitmap 'bitmap_data1' to file: Invalid argument
01:55:54 INFO | [qemu output] (Process terminated with status 0)

1. is this expected? if it's expected, I'd like to remove nbd backend from the case
2. if this is an issue, I think we can still merge the patch for it is a nbd-specific issue.

**Comment by zhencliu** [comment]
hi @YongxueHong @ngu-niny , can you please help to review this patch? Thanks.

**Review by YongxueHong — APPROVED** [review]
LGTM.

File: qemu/tests/blockdev_inc_backup_rm_persistent_bitmap.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    Persistent bitmaps removal testing
+
+    test steps:
+        1. boot VM with a 2G data disk
+        2. format data disk and mount it, create a file
+        3. add target disks for backup to VM via qmp commands
+        4. do full backup and add persistent bitmap
+        5. create another file
+        6. disable bitmaps, record the count of bitmaps
+        7. shutdown&restart vm, check count of bitmaps should keep the same
```
Comment (ngu-niny) [inline qemu/tests/blockdev_inc_backup_rm_persistent_bitmap.py:68]: There is no bitmap check after restart the VM, shall we add it?

Comment (zhencliu) [reply qemu/tests/blockdev_inc_backup_rm_persistent_bitmap.py]: It used to check the bitmaps after vm restart, then I removed the check for #2552 covered this part. Just updated the test steps, please make a double check, thanks

File: qemu/tests/blockdev_inc_backup_rm_persistent_bitmap.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    Persistent bitmaps removal testing
+
+    test steps:
+        1. boot VM with a 2G data disk
+        2. format data disk and mount it, create a file
+        3. add target disks for backup to VM via qmp commands
+        4. do full backup and add persistent bitmap
+        5. create another file
+        6. disable bitmaps, record the count of bitmaps
+        7. poweroff&restart vm
```
Comment (ngu-niny) [inline qemu/tests/blockdev_inc_backup_rm_persistent_bitmap.py:68]: @zhencliu Hi Zhenchao, I have not express clearly, suggest to add bitmap check both in step5 and step7. How do you think?

Comment (zhencliu) [reply qemu/tests/blockdev_inc_backup_rm_persistent_bitmap.py:68]: It's OK for me, let's make it clear:
after step 5: count should be greater than 0
after step 7: count should keep the same and status is disabled

**Comment by zhencliu** [comment]
(1/1) type_specific.io-github-autotest-qemu.blockdev_inc_backup_rm_persistent_bitmap: PASS (108.86 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-12-17T01.35-f92991c/results.html
JOB TIME   : 111.34 s

(1/1) type_specific.io-github-autotest-qemu.blockdev_inc_backup_rm_persistent_bitmap: PASS (143.05 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-12-17T01.52-c78ba2b/results.html
JOB TIME   : 145.81 s

hi @ngu-niny , change done, added bitmaps check, please help to make a double check, thanks.

**Comment by ngu-niny** [comment]
LGTM and test PASS:

(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_rm_persistent_bitmap: PASS (210.75 s)


Acked-by: Nini Gu <ngu@redhta.com>

**Pull request merged**